service/src/main/java/uk/nhs/adaptors/oneoneone/cda/report/mapper/ReferralRequestMapper.java

```diff
@@ -0,0 +1,73 @@
+package uk.nhs.adaptors.oneoneone.cda.report.mapper;
+
+import lombok.RequiredArgsConstructor;
+import org.hl7.fhir.dstu3.model.*;
+import org.springframework.stereotype.Component;
+import uk.nhs.adaptors.oneoneone.cda.report.service.HealthcareServiceService;
+
+import uk.nhs.connect.iucds.cda.ucr.*;
+
+import java.util.Date;
+
+import java.util.Base64;
+
+
+@Component
+@RequiredArgsConstructor
+public class ReferralRequestMapper {
+
+    private final HealthcareServiceService healthcareServiceService;
+    private final PatientMapper patientMapper;
+    private final EncounterMapper encounterMapper;
+
+    public ReferralRequest transform(POCDMT000002UK01ClinicalDocument1 clinicalDocument) {
+
+        //TODO: Pathways - to discuss with team
+
+//        String s = clinicalDocument.getComponent().getStructuredBody().getComponentArray(0)
+//                .getSection().getEntryArray(0).getObservationMedia().getValue().xmlText();
+//
+//        int startPosition = s.indexOf(">\n                ") + ">\n                ".length();
+//        int endPosition = s.indexOf("<", startPosition);
+//        String subS = s.substring(startPosition, endPosition);
+//
+//        byte[] decoded = Base64.getDecoder().decode(subS.getBytes());
+
+        ReferralRequest referralRequest = new ReferralRequest();
+        POCDMT000002UK01PatientRole patient = clinicalDocument.getRecordTargetArray(0).getPatientRole();
```

Comment:
clinicalDocument.getRecordTargetArray(0).getPatientRole() -> this also to encounter.subject, please contact with @babasholaking-kainos